extensions/omni_python/docs/intro.md

```diff
@@ -0,0 +1,196 @@
+```python
+from omni_python import pg
+
+@pg
+def hello() -> str:
+    return "Hey there!"
+```
+
+`maths.py`:
+```python
+from omni_python import pg
+
+@pg
+def add(a: int, b: int) -> int:
+    return a + b
+
+@pg
+def subtract(a: int, b: int) -> int:
+    return a - b
+```
+
+4. Create a function to define an identifier for the local directory you want to load into the database. For the example, the local directory is `python-files`.
+
+```postgresql
+create or replace function demo_function() returns omni_vfs.local_fs language sql
+as $$
+select omni_vfs.local_fs('/python-files')
+$$;
+```
+
+5. Configure `omni_python`
+```postgresql
+insert into omni_python.config (name, value) values ('pip_find_links', '/python-wheels');
+```
+
+6. Load the filesystem files.
+```postgresql
+select omni_schema.load_from_fs(demo_function());
+```
+
+7. [Optional] You can set a reload command for reloading the filesystem changes.
+```postgresql
+\set reload 'select omni_schema.load_from_fs(demo_function());'
+```
+
+8. Run Omnigres in a Docker container. Make sure to mount the local directory as a volume on the correct path.
+```shell
+docker run --name omnigres \
+           -e POSTGRES_PASSWORD=omnigres \
+           -e POSTGRES_USER=omnigres \
+           -e POSTGRES_DB=omnigres \
+           --mount source=omnigres,target=/var/lib/postgresql/data -v $(pwd)/python-files:/python-files \
+           -p 127.0.0.1:5433:5432 --rm ghcr.io/omnigres/omnigres-slim:latest
+```
+
+9. Let's try it out!
+```postgresql
+omnigres=# select hello();
+   hello
+------------
+ Hey there!
+```
+```postgresql
+omnigres=# select add(5, 10);
+ add
+-----
+  15
+```
+```postgresql
+omnigres=# select subtract(5, 10);
+ subtract
+----------
+       -5
+```
+
+## Flask
+
+For Flask framework integration, we have a few more steps.
+
+1. Create `omni_httpd` extension to be able to handle HTTP requests.
+```postgresql
+create extension if not exists omni_httpd cascade;
+```
+2. Add the following in `requirements.txt` file
+```
+omni_http[Flask]
+```
+
+3. Create HTTP listeners for our Flask application. For example, let's add a HTTP listener for port 5000.
+```postgresql
+with
+    listener as (insert into omni_httpd.listeners (address, port) values ('0.0.0.0', 5000) returning id),
+    handler as (insert into omni_httpd.handlers (query) values
+                                                 (
+                                                 $$ select handle(request.*) from request$$
+       ) returning id)
+insert
+into
+    omni_httpd.listeners_handlers (listener_id, handler_id)
+select
+    listener.id,
+    handler.id
+from
+    listener,
+    handler;
+```
+
+4. Now you can update your Python files (in the mounted volume) to include Flask functionality.
+```python
+from omni_python import pg
+from omni_http import omni_httpd
+from omni_http.omni_httpd import flask
+from flask import Flask, jsonify, make_response
+
+app = Flask('myapp')
+
+@app.route("/")
+def ping():
```

Comment:
I read in some past thread that we don't have the DB API ready for Flask right now?

I was initially thinking of having the Flask app store some DB record and fetch it, any hints on that?

Or any other suggestions on what kind of examples you'd like to have?

I felt like the examples you demonstrated showcase the basic functionalities quite nicely, so re-used those.

Reply:
This is correct that we do not have DB API support just yet (this is completely orthogonal to Flask integration) and of course, we want that ASAP.

But, in the short term, we can still use [plpython's own API for database access](https://www.postgresql.org/docs/current/plpython-database.html) and perhaps leave a comment there saying that this is to be replaced by DB API compatible APIs and/or other frameworks (such as SQLAlchemy) when this will be ready, to give people an idea of where this is heading.

Reply:
As for examples, to me they feel messy because they were literally meaningless "somethings" I quickly put together.  If we can create a super small mini app that does something useful (write something to the database and read back in another point, or something to that end), that'd be fantastic.

To make sure it really works we need to get #316 in as it fixes rather important issues.

Reply:
@yrashk Have added a better example, and left todo comment where we had a dependency on the open PR. Can take it up separately later.
Hope that works!